### ISSUE TITLE
[STORM-972] Metrics logging frequency should not be affected with bolt execute slowness

### ISSUE THREAD

**Issue by jira-importer** [apache/storm#4810]
<p>I had registered metrics in my bolt for 60 seconds, but I noticed that they were not getting logged even after two minutes. After running the topology for a while, bolt metrics were logged eventually but not according to the metricBucketSizeInSecs. System inbuilt metrics were working just fine. </p>

<p>Bolt actually sends the metrics to the consumer bolt, only after it receives the metric tick tuple for processing. Even though the tick was generated after 60 seconds, it is sent in the same queue along with messages from spout. Bolt will see the tick only after processing messages from spout, which could be any duration. </p>

<p>If this diagnosis is correct, can we just for metric interval lapse in a loop instead of relying on metric tick. I suppose this problem only exists for the bolt.  </p>

---
<details><summary><i>Originally reported by <a title="Abhishek Agarwal" href="https://issues.apache.org/jira/secure/ViewProfile.jspa?name=abhishek.agarwal">abhishek.agarwal</a>, imported from: <a href="https://issues.apache.org/jira/browse/STORM-972" target="_blank">Metrics logging frequency should not be affected with bolt execute slowness</a></i></summary>
<i><ul>
<li><b>status</b>: Resolved
<li><b>priority</b>: Major
<li><b>resolution</b>: Won't Fix
<li><b>resolved</b>: 2016-02-15T13:42:35+00:00
<li><b>imported</b>: 2025-01-24
</ul></i>
</details>
